ipfshttpclient/filescanner.py

```diff
@@ -656,15 +636,15 @@ def _walk(
 					# Report the target file or directory
 					if is_dir:
 						reported_directories.add(filepath)
-						yield FSNodeEntry(  # type: ignore[misc]  # mypy bug: gh/python/mypy#685
```

Comment:
I have a suspicion that the 'bugs' are not actually bugs, but rather problems with the type declarations.

I made a significant effort (not included in the PR) to resolve the type errors after removing some of the warnings. There were quite a few challenges in `file_scanner` that seem to suggest that there's problems being covered up with hints, which causes more problems, and thus more hints.

I ended up rolling back that refactor because it was a deep rabbit hole. I'm going to make another stab at it and see if I can offer a more educated commentary once I've better formed my own view on what may be going on.

Reply:
Unfortunately I've had a somewhat different experience though: There are things that feel like they should be easy to describe, but no matter how hard you try you always just fall short of trying to teach mypy what is plainly obvious to you. Eventually you realize you have exhausted the possible solution space and find out that that particular thing you're to describe happens to be something that mypy doesn't quite support yet or gets wrong.

So please, for your own sanity, do yourself a favour and don't waste too much time at trying to figure this out. If I'm citing a mypy bug number it most likely means I already went through that ordeal and found out that that particular issue is indeed the problem, even if it doesn't appear to make sense at first.

In this case the “error” however is indeed that `FSNodeEntry` contains generic fields (`ty.AnyStr`) and mypy cannot deal with it (the linked issue).